contracts/lib/LibPerpetuals.sol

```diff
@@ -114,11 +115,14 @@ library Perpetuals {
         bool opposingSides = a.side != b.side;
         // long order must have a price >= short order
         bool pricesMatch = a.side == Side.Long ? a.price >= b.price : a.price <= b.price;
+        bool marketsMatch = a.market == b.market;
+        bool makersDifferent = a.maker != b.maker;
```

Comment:
I'm wondering if this is important? If a user makes an order that gets successfully matched against another order made by themself, through the order book, wouldn't expected behaviour be to match them?
@sporejack wonder what you think about this.

Reply:
@BenjaminPatch we discussed on Friday, I would say this is just a sanity check to avoid an manipulation on low liquidity books. Obviously the easiest way around this is just have two different accounts make the order.

If we treat the contracts as the absolute truth, it makes sense for them to check this predicate imo

Reply:
Yeah but as you said it's trivial to circumvent it by using two different accounts

Reply:
Yeah good point, I'll get Jacks opinion on this and we can make a decision. An extra check that isn't that valuable is kinda pointless.

Reply:
There were several defects found which would be mitigated by disallowing trading with oneself; however, given the refactor these are quite possibly mitigated entirely now. Provided `TCR-07` is fixed, wash trading shouldn't impact on-chain data in an adverse way. 

Conversely, what's the realistic downside to having this check in place? Aside from a UX perspective, gas is likely to be negligible (we're comparing two in-memory variables). Open to thoughts on this from both of you @raymogg and @BenjaminPatch .

Reply:
If you think it's safer for this check, I'm fine with it. I guess we just have to make sure the OME also does this same check to make sure invalid order pairs aren't put on chain (I assume it already does)

Either way I'm easy